tests/integration/components/language-picker/component-test.js

```diff
@@ -0,0 +1,31 @@
+import { moduleForComponent, test } from 'ember-qunit';
+import hbs from 'htmlbars-inline-precompile';
+import Ember from 'ember';
+
+
+const localeStub = Ember.Service.extend({
+  locale: null
+});
+
+moduleForComponent('language-picker', 'Integration | Component | language picker', {
+  integration: true,
+
+  beforeEach: function() {
+    this.register('service:i18n', localeStub);
+    this.inject.service('i18n', { as: 'i18n' });
+  }
+});
+
+test('it sets the locale', function(assert) {
+
+  this.set('selectLanguage', (language, code) => {
+    assert.deepEqual(language, 'English (US)');
+    assert.deepEqual(code, 'en-US');
+  });
+
+  this.render(hbs`{{language-picker onPick=(action selectLanguage)}}`);
```

Comment:
So is this just vetting that the action is called with specific arguments?

One thing I would suggest- any time that some of your assertions are buried inside a function call that is conditional on another event, you want to make sure that ember has actually fired those assertions during the course of the test.

(Otherwise someone could remove the action from the button in the template. Your tests would seem to pass when in fact they weren't even being run.)

You can tell QUnit how many assertions should run using `assert.expects`.
https://api.qunitjs.com/expect/